core/arch/arm/mm/tee_mmu.c

```diff
@@ -370,8 +370,10 @@ static TEE_Result split_vm_region(struct user_mode_ctx *uctx,
 	if (mobj_is_paged(r->mobj)) {
 		TEE_Result res = tee_pager_split_um_region(uctx, va);
 
-		if (res)
+		if (res != TEE_SUCCESS) {
+			free(r2);
```

Comment:
Good catch also.
Here, prefer keep `if (res)` instead of `if (res != TEE_SUCCESS)`.

Reply:
So optee coding style allowing enums to be treated as essential bools?

Reply:
Yes, `if (foo)` can be used with `foo` being a boolean, a pointer or a numerical value (even  an enumerated value).

By definition `TEE_SUCCESS` is zero. That's in the GPD TEE spec OP-TEE is based on. Using `if (res)` instead of `if (res != TEE_SUCCESS)` and `if (!res)` instead of `if (res == TEE_SUCCESS)` make to code easier to read since we really often test `TEE_Result` value against `TEE_SUCCESS`.

Note: optee_os coding style is mostly based on the Linux kernel coding style.

Reply:
res != TEE_SUCCESS exists in 665 places since before

if (res) existed in 430 places since before

Iv changed to if (res)

Reply:
there seems to be a mix in the codebase with about 60% res != TEE_SUCCESS and 40% if (res) - if changed anyways